src/models/v2/asyncapi.ts

```diff
@@ -76,7 +77,16 @@ export class AsyncAPIDocument extends BaseModel<v2.AsyncAPIObject> implements As
   }
 
   schemas(): SchemasInterface {
-    return new Schemas([]);
+    const schemas: Set<SchemaInterface> = new Set();
+    function callback(schema: SchemaInterface) {
+      if (!schemas.has(schema.json())) {
+        schemas.add(schema);
+      }
+    }
+    // return only schemas used in channels ("active" schemas)
```

Comment:
This is raising a hidden behaviour I didn't think when designing the API. Are we sure this is the behavior all users will need? Once i have the "active" schemas, how do I know the ones are not active?

I think we should rather return **all schemas** and add a filter in the `Schemas` collection such as `FilterByActive()` and `FilterByInactive` or `FilterByInUse()`/`FilterByNotInUse()` or whatever.

Reply:
If someone will need also inactive schemas can write:

```ts
    const schemas: Set<SchemaInterface> = new Set();
    function callback(schema: SchemaInterface) {
      if (!schemas.has(schema.json())) {
        schemas.add(schema);
      }
    }
    traverseAsyncApiDocument(this, callback);
    const schemas = new Schemas(Array.from(schemas));
```

Reply:
Sure they can but I don't think we should take this decision on return only the ones used by channels but rather all of them. Then, thanks to filters, the could filter after all.

Reply:
🤷🏼

Reply:
We had a chat on Slack about it. As it ended up making sense, I'm gonna create those filters not only for `Schemas` but also for `Channels`, `Operations`, `Messages`, and `Servers`, also a PR to `parser-api` reflecting the change.
Can you @magicmatatjahu then make this method to return all schemas?

Reply:
https://github.com/asyncapi/parser-js/pull/619

Reply:
@magicmatatjahu would you have some time to review the PR mentioned in my previous component, so we can unblock this? 🙏

Reply:
I think you can adapt the code of this PR to consider https://github.com/asyncapi/parser-js/pull/619 is gonna happen so we can merge this right? It is not really blocked in fact.

Reply:
https://github.com/asyncapi/parser-js/pull/619#discussion_r985692933